internal/scheduler/jobiteration.go

```diff
@@ -51,6 +52,8 @@ type InMemoryJobRepository struct {
 	// If true, jobs are sorted first by priority class priority.
 	// If false, priority class is ignored when ordering jobs.
 	sortByPriorityClass bool
+	// Protects the above fields.
+	mu sync.Mutex
```

Comment:
I had a test fail due to concurrent map modification in this class so I've added a lock.